hack/update-codegen.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+command -v deepcopy-gen >/dev/null 2>&1 || {
```

Comment:
FYI it's probably easier to vendor it and then run it from the vendored directory :-)

Reply:
I absolutely agree in terms of UX. The concern here is that the code-generator tools add extra dependencies that I don't think I need. But maybe there's a way to say: hey I need this for tooling / tests but you don't need to pull this in as a consumer of this library?